src/main/java/StudentSystem/Controller/CourseController.java

```diff
@@ -18,9 +21,11 @@
 public class CourseController {
 
     private CourseRepository courseRepository;
+    private DepartmentRepository departmentRepository;
 
-    public CourseController(CourseRepository courseRepository) {
+    public CourseController(CourseRepository courseRepository, DepartmentRepository departmentRepository) {
```

Comment:
Instead of having constructors, try using the @Autowired annotation. The biggest feature of Spring is the dependency injection and you should try to use it at most times.